web/components/portfolio/portfolio-value-graph.tsx

```diff
@@ -27,37 +34,138 @@ export const PortfolioTooltip = (props: { date: Date }) => {
 export const PortfolioGraph = (props: {
   mode: GraphMode
   duration?: Period
-  points: HistoryPoint<Partial<PortfolioMetrics>>[]
+  portfolioHistory: PortfolioSnapshot[]
   width: number
   height: number
   zoomParams?: ZoomParams
-  onMouseOver?: (p: HistoryPoint<Partial<PortfolioMetrics>> | undefined) => void
   negativeThreshold?: number
   hideXAxis?: boolean
+  firstProfit: number
+  setGraphBalance: (balance: number | undefined) => void
+  setGraphInvested: (invested: number | undefined) => void
+  setGraphProfit: (profit: number | undefined) => void
+  portfolioFocus: PortfolioMode
+  setPortfolioFocus: (mode: PortfolioMode) => void
+  portfolioHoveredGraph: PortfolioHoveredGraphType
+  setPortfolioHoveredGraph: (hovered: PortfolioHoveredGraphType) => void
 }) => {
   const {
     mode,
     duration,
-    points,
-    onMouseOver,
+    // points,
+    firstProfit,
+    portfolioHistory,
     width,
     height,
     zoomParams,
-    negativeThreshold,
+    negativeThreshold = 0,
     hideXAxis,
+    setGraphBalance,
+    setGraphInvested,
+    setGraphProfit,
+    portfolioFocus,
+    setPortfolioFocus,
+    portfolioHoveredGraph,
+    setPortfolioHoveredGraph,
   } = props
+
+  const { profitPoints, investmentPoints, balancePoints, networthPoints } =
+    useMemo(() => {
+      if (!portfolioHistory?.length) {
+        return {
+          profitPoints: [],
+          investmentPoints: [],
+          balancePoints: [],
+          networthPoints: [],
+        }
+      }
+
+      const profitPoints: HistoryPoint<Partial<PortfolioMetrics>>[] = []
+      const investmentPoints: HistoryPoint<Partial<PortfolioMetrics>>[] = []
+      const balancePoints: HistoryPoint<Partial<PortfolioMetrics>>[] = []
+      const networthPoints: HistoryPoint<Partial<PortfolioMetrics>>[] = []
+
+      portfolioHistory.forEach((p) => {
+        profitPoints.push({
+          x: p.timestamp,
+          y: p.balance + p.investmentValue - p.totalDeposits - firstProfit,
+          obj: p,
+        })
+        investmentPoints.push({
+          x: p.timestamp,
+          y: p.investmentValue,
+          obj: p,
+        })
+        balancePoints.push({
+          x: p.timestamp,
+          y: p.balance,
+          obj: p,
+        })
+        networthPoints.push({
+          x: p.timestamp,
+          y: p.balance + p.investmentValue,
+          obj: p,
+        })
+      })
+
+      return { profitPoints, investmentPoints, balancePoints, networthPoints }
```

Comment:
This might be nice to move into `common` as a utility function. Or else at the bottom of the file.